src/main/java/com/yahoo/bullet/operations/aggregations/sketches/QuantileSketch.java

```diff
@@ -0,0 +1,287 @@
+
+        addIfKeyNonNull(metadata, conceptKeys.get(Concept.MINIMUM_VALUE.getName()), this::getMinimum);
+        addIfKeyNonNull(metadata, conceptKeys.get(Concept.MAXIMUM_VALUE.getName()), this::getMaximum);
+        addIfKeyNonNull(metadata, conceptKeys.get(Concept.ITEMS_SEEN.getName()), this::getNumberOfEntries);
+        addIfKeyNonNull(metadata, conceptKeys.get(Concept.NORMALIZED_RANK_ERROR.getName()), this::getNormalizedRankError);
+
+        return metadata;
+    }
+
+    @Override
+    protected Boolean isEstimationMode() {
+        return merged.isEstimationMode();
+    }
+
+    @Override
+    protected String getFamily() {
+        return Family.QUANTILES.getFamilyName();
+    }
+
+    @Override
+    protected Integer getSize() {
+        return merged.getStorageBytes();
+    }
+
+    private Double getMinimum() {
+        return merged.getMinValue();
+    }
+
+    private Double getMaximum() {
+        return merged.getMaxValue();
+    }
+
+    private Long getNumberOfEntries() {
+        return merged.getN();
+    }
+
+    private Double getNormalizedRankError() {
+        return merged.getNormalizedRankError();
+    }
+
+    private double[] getDomain() {
+        if (numberOfPoints != null) {
+            return type == DistributionType.QUANTILE ? getPoints(QUANTILE_MIN, QUANTILE_MAX, numberOfPoints) :
+                                                       getPoints(getMinimum(), getMaximum(), numberOfPoints);
+        }
+        return points;
+    }
+
+    /**
+     * Exposed for testing only.
+     *
+     * Creates a {@link List} of {@link BulletRecord} for each corresponding entry in domain and range. The domain
+     * is first converted into range names depending on the type.
+     *
+     * @param domain An array of split points of size N > 0.
+     * @param range  An array of values for each range in domain: of size N + 1
+     *               if type is not {@link DistributionType#QUANTILE} else N.
+     * @param type The {@link DistributionType} to zip for.
+     * @param n A long to scale the value of each range entry by if type is not {@link DistributionType#QUANTILE}.
+     * @return The records that correspond to the data.
+     */
+    static List<BulletRecord> zip(double[] domain, double[] range, DistributionType type, long n) {
+        List<BulletRecord> records = null;
+        switch (type) {
+            case QUANTILE:
+                records = zipQuantiles(domain, range);
+                break;
+            case PMF:
+                records = zipRanges(domain, range, n, false);
+                break;
+            case CDF:
+                records = zipRanges(domain, range, n, true);
+                break;
+        }
+        return records;
+    }
+
+    // Static helpers
+
+    private static double[] getPoints(double start, double end, int numberOfPoints) {
+        double[] points = new double[numberOfPoints];
+
+        // We should have points < 1 but just in case...
```

Comment:
Also, since I do this check I should initialize the array after. It won't happen because the validations have already been done in Distribution but since I check here. I should do that too. Will make that change.